client/state/media/reducer.js

```diff
@@ -36,6 +40,52 @@ export const items = createReducer( {}, {
 	}
 } );
 
+export const queries = ( () => {
+	function applyToManager( state, siteId, method, createDefault, ...args ) {
```

Comment:
Maybe I should have glanced at the changes before writing my initial comment 😆  But in this case I think we don't need both `items` and `queries`. A `QueryManager` instance allows us to retrieve a single item, all items, or items by queries, which should satisfy all of our selector requirements.